### PR TITLE
Implemented Software Stack and OS Image Create

### PR DESCRIPTION
**Pull request by srikanthv02**
@udeshrl  : Please review.

### CODE REVIEW

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
+(function (angular) {
+    "use strict";
+    angular.module('dashboard.design')
+        .controller('blueprintCreateCtrl',['$scope','$rootScope','$http','$q','toastr','$state','designServices', 'blueprintCreateService','confirmbox', function ($scope,$rootScope,$http,$q,toastr,$state,desServ,bpCreateSer,confirmbox) {
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Remove the dependency injection , which are not used inside the controller.

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+            var blueprintCreation = this;
+            //to get the templates listing.
+            $scope.bpType = $state.params.templateObj.templatetypename;
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: var name is "bpType " but value is "templatetypename" . Just change to "bpTypeName"

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+            blueprintCreation.buProjListing = [];
+            blueprintCreation.projListing = [];
+
+            blueprintCreation.templateListing = function(){
+                bpCreateSer.getTemplates().then(function(data){
+                    $scope.templateList = data;    
+                });
+            };
+
+            $scope.blueprintTemplateClick = function(templateDetail) {
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: When user select any template , at that time below API will call .

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+            blueprintCreation.getCFTParams = function() {
+                var cftTemplate = $scope.templateSelected;
+                cftTemplate = $scope.templateSelected.rowid + "__template__" + $scope.templateSelected.template_filename;
+                bpCreateSer.getCFTParams(cftTemplate).then(function(data){
+                    $scope.getCFTDetails = data;
+                });
+            };                                             
+
+            //wizard data setting for step 1 and step 2.
+            var index = 0, // points to the current step in the steps array
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Create a method for this wizard. And use it .

File: client/cat3/src/partials/sections/dashboard/design/services/blueprintCreateServices.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+					url: '/aws/providers/vpc/'+ vpcId +'/securitygroups',
+					data: {
+	                    "providerId": providerId,
+	                    "region": region
+                	}
+				};
+				return designServices.promisePost(params);
+			};
+			//listing down the subnets based upon the VPC ID.(set the instance count to 10 from Ctrl)
+			blueprintServices.getOrgBuProj = function () {
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/services/blueprintCreateServices.js]: All ready  called this API .

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+                                 <option ng-repeat="config in bpCreate.projListing" value="{{config.businessGroups[0].projects[0].rowid}}">{{config.businessGroups[0].projects[0].name}}</option>
+                              </select>
+                           </div>
+                           
+                           <!-- Docker Image selection table -->
+                           <div ng-if="templateSelected.templatetypename == 'Docker'" class="col-lg-12 col-md-12" style="margin-top:9px;">
+                              <label for="category" style="margin-top:5px;">Docker Images<span class="red">&nbsp;*</span>
+                              </label>
+                              <a class="btn btn-primary adddockertemplate pull-right" onclick="showdockertemplateadder();" style="margin-bottom:5px;">Add</a>
+                              <table datatable="" class="table-bordered text-center">
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: In angular not used  " datatable ".

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html [hunk 7]
```diff
@@ -0,0 +1,488 @@
...
+                              </span>
+                              <div class="view-runlist">
+                                 <table class="table table-bordered text-center">
+                                    <thead>
+                                       <tr>
+                                          <th class="text-center">View Runlist</th>
+                                       </tr>
+                                    </thead>
+                                    <tbody>
+                                       <tr ng-repeat="item in templateSelected.templatescookbooks.split(',')">
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: why do you have to use  "split(',')" ?

File: client/cat3/src/partials/sections/dashboard/genericServices.js [hunk 8]
```diff
@@ -82,5 +82,27 @@
                 });
             }
         };
+        genericServices.editRunlist = function(chefRunlist, chefAttribute) {
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/genericServices.js]: "genericServices.editRunlist" is not using any place . Just remove it .

File: client/cat3/src/partials/sections/dashboard/genericServices.js [hunk 9]
```diff
@@ -82,5 +82,27 @@
...
+                            attributes: chefAttribute
+                        };
+                    }
+                }
+            }).result.then(function (selectedCookBooks) {
+                //$rootScope.editRunListAttributes = false;
+               // $rootScope.chefrunlist = selectedCookBooks.list;
+               // $rootScope.cookbookAttributes = selectedCookBooks.cbAttributes;
+            }, function () {
+                console.log('Dismiss time is ' + new Date());
```
Comment (hrushikesh07) [inline client/cat3/src/partials/sections/dashboard/genericServices.js]: Remove the console.log().

File: client/cat3/src/partials/sections/dashboard/genericServices.js (hunk 8 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/genericServices.js]: genericServices.editRunlist - Development is under process. Have commented it for now. Using the split for the same reason as the runlists are coming in string format(comma separated values).

File: client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html (hunk 6 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/design/view/blueprintCreate.html]: Done.

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js (hunk 4 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Will do that in the next PR.

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js (hunk 3 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Done

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js (hunk 2 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Done

File: client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js (hunk 1 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/design/controllers/blueprintCreateCtrl.js]: Done

File: client/cat3/src/partials/sections/dashboard/design/services/blueprintCreateServices.js (hunk 5 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/design/services/blueprintCreateServices.js]: Will remove the api once the re-factor is done.

File: client/cat3/src/partials/sections/dashboard/genericServices.js (hunk 9 above)
Comment (srikanthv02) [inline client/cat3/src/partials/sections/dashboard/genericServices.js]: Removed

**Comment by srikanthv02** [comment]
@udeshrl : PR has been updated with the review comments.

**Pull request merged**